### ISSUE TITLE
molecule matrix

### ISSUE THREAD

**Issue by rgarrigue** [metacloud/molecule#1426]
# Issue Type

- Bug report

# Molecule and Ansible details

```
 ansible --version; molecule --version
 [WARNING] Ansible is in a world writable directory (/vagrant/ansible-role-robotframework), ignoring it as an ansible.cfg source.
ansible 2.6.2
  config file = None
  configured module search path = ['/home/vagrant/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.6/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.6.5 (default, Apr 10 2018, 17:08:37) [GCC 4.8.5 20150623 (Red Hat 4.8.5-16)]
molecule, version 2.16.0
```

Molecule installation method (one of):

- pip

Ansible installation method (one of):

- pip

# Actual Behaviour (Bug report only)

Trying to understand and change the test matrix, I tried the `molecule matrix` subcommand without args, and boom python error stack. It should just send back usage.

```
[vagrant@ansible ansible-role-robotframework]$ molecule matrix
Traceback (most recent call last):
  File "/usr/bin/molecule", line 11, in <module>
    sys.exit(main())
  File "/usr/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/molecule/command/matrix.py", line 76, in matrix
    'subcommand': subcommand[0],
IndexError: tuple index out of range
```

And imho the usage (or the doc) could be cleared about what's a subcommand. 

```
Usage: molecule matrix [OPTIONS] [SUBCOMMAND]...

  List matrix of steps used to test instances.

Options:
  -s, --scenario-name TEXT  Name of the scenario to target.
  --help                    Show this message and exit.
```

**Comment by lwm** [comment]
Duplicate of https://github.com/metacloud/molecule/issues/1399 which was fixed by https://github.com/metacloud/molecule/pull/1410?

**Comment by retr0h** [comment]
Will be released w/ 2.17.
